### ISSUE TITLE
[Clang][C++26] Raw string literals with newly perimitted d-chars are still rejected

### ISSUE THREAD

**Issue by frederick-vs-ja** [llvm/llvm-project#93130]
https://clang.llvm.org/cxx_status.html currently claims that [P2558R2](https://wg21.link/p2558r2) is implemented. However, per [[lex.string]](https://eel.is/c++draft/lex.string), `@`, `$`, and `` ` `` need to be usable as d-chars since C++26, and Clang doesn't accept the newly permitted raw string literals yet.

[Godbolt link](https://godbolt.org/z/W54b1heq4)
```C++
int main() {
  (void) R"abc`@$(foobar)abc`@$";
}
```

Perhaps we should add pedantic warning/error for these permitted raw string literals in pre-C++26 modes.

Originally discovered by @jakubjelinek in https://gcc.gnu.org/PR110343.

**Comment by romanova-ekaterina** [comment]
Quick question: 

This change (allow `@$ in raw string delimiters) is supposed to be applicable only to the C++26 standard, right? 
Consider the testcase below. No diagnostic is reported with the latest clang compiler even if I compile with C++11 standard. 

clang.exe  -c -Wall -Wpedantic -std=c++11 test.cpp 

int main()
{
	char		str[] = R"$(\no)$"; /* Diagnostic is expected here */
        return 0;
}

I suspect it has not been done on purpose and regression accidentally has been introduced.

Older clang compiler (as well as msvc and gcc compiler) report an error for this testcase.

Would you like me to file a bug?
